thrall/app/controllers/ReaperController.scala

```diff
@@ -0,0 +1,196 @@
+)(implicit val ec: ExecutionContext) extends BaseControllerWithLoginRedirects with GridLogging {
+
+  private val CONTROL_FILE_NAME = "PAUSED"
+
+  private val INTERVAL = 15 minutes // based on max of 1000 per reap, this interval will max out at 96,000 images per day
+  private val REAPS_PER_WEEK = 7.days / INTERVAL
+
+  implicit val logMarker: MarkerMap = MarkerMap()
+
+  private def getIsReapable = new ReapableEligibility {
+    override val persistedRootCollections: List[String] = config.persistedRootCollections
+    override val persistenceIdentifier: String = config.persistenceIdentifier
+  }
+
+  config.maybeReaperBucket.foreach { reaperBucket =>
+    scheduler.scheduleAtFixedRate(
+      initialDelay = 0 seconds,
+      interval = INTERVAL,
+    ){ () =>
+      if(store.client.doesObjectExist(reaperBucket, CONTROL_FILE_NAME)) {
+        logger.info("Reaper is paused")
+      } else {
+        es.countImagesIngestedInLast(7 days)(DateTime.now(DateTimeZone.UTC)).flatMap { imagesIngestedInLast7Days =>
+
+          val imagesIngestedPer15Mins = imagesIngestedInLast7Days / REAPS_PER_WEEK
+
+          val countOfImagesToReap = Math.min(imagesIngestedPer15Mins, 1000).toInt
+
+          if (countOfImagesToReap == 1000) {
+            logger.warn(s"Reaper is reaping at maximum rate of 1000 images per $INTERVAL. If this persists, the INTERVAL will need to become more frequent.")
+          }
+
+          val isReapable = getIsReapable
+
+          Future.sequence(Seq(
+            doBatchSoftReap(countOfImagesToReap, deletedBy = "reaper", isReapable),
+            doBatchHardReap(countOfImagesToReap, deletedBy = "reaper", isReapable)
+          ))
+        }
+      }
+    }
+  }
+
+  private def batchDeleteWrapper(count: Int)(func: (Int, String, ReapableEligibility) => Future[JsValue]) = auth.async { request =>
+    if (!authorisation.hasPermissionTo(DeleteImage)(request.user)) {
+      Future.successful(Forbidden)
+    }
+    else if (count > 1000) {
+      Future.successful(BadRequest("Too many IDs. Maximum 1000."))
+    }
+    else {
+      func(
+        count,
+        request.user.accessor.identity,
+        getIsReapable
+      ).map(Ok(_))
+    }
+  }
+
+  private def persistedBatchDeleteOperation(deleteType: String)(doBatchDelete: => Future[JsValue]) = config.maybeReaperBucket match {
+    case None => Future.failed(new Exception("Reaper bucket not configured"))
+    case Some(reaperBucket) => doBatchDelete.map { json =>
+      val now = DateTime.now(DateTimeZone.UTC)
+      val key = s"$deleteType/${now.toString("YYYY-MM-dd")}/$deleteType-${now.toString()}.json"
+      store.client.putObject(reaperBucket, key, json.toString())
+      json
+    }
+  }
+
+  def doBatchSoftReap(count: Int): Action[AnyContent] = batchDeleteWrapper(count)(doBatchSoftReap)
+
+  def doBatchSoftReap(count: Int, deletedBy: String, isReapable: ReapableEligibility): Future[JsValue] = persistedBatchDeleteOperation("soft"){
+
```

Comment:
optional for this PR, but I'd suggest building up a logMarker with the request ID (example https://github.com/guardian/grid/blob/main/cropper/app/controllers/CropperController.scala#L55) and passing it around and feeding it into the log statements; may make life easier tracking what happened in a given request. (in normal operation there won't be much ambiguity because calls are 15 mins apart, but it would allow you to quickly filter away any other thrall logs, or useful if we make a bunch of calls quickly to manually clear a backlog)